tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/CommentExtractor.cpp

```diff
@@ -0,0 +1,875 @@
+          cast<const comments::VerbatimBlockComment>(comment));
+    case comments::Comment::CommentKind::InlineCommandCommentKind:
+      return std::make_unique<AstInlineCommand>(
+          cast<const comments::InlineCommandComment>(comment));
+    case comments::Comment::ParagraphCommentKind:
+      return std::make_unique<AstParagraphComment>(cast<const comments::ParagraphComment>(comment));
+    case comments::Comment::TextCommentKind:
+      return std::make_unique<AstTextComment>(cast<const comments::TextComment>(comment));
+    case comments::Comment::VerbatimBlockLineCommentKind:
+      return std::make_unique<AstVerbatimBlockLineComment>(
+          cast<const comments::VerbatimBlockLineComment>(comment));
+    case comments::Comment::VerbatimLineCommentKind:
+      return std::make_unique<AstVerbatimLineComment>(
+          cast<const comments::VerbatimLineComment>(comment));
+
+    case comments::Comment::HTMLStartTagCommentKind:
+      return std::make_unique<AstHtmlStartTagComment>(
+          cast<const comments::HTMLStartTagComment>(comment));
+    case comments::Comment::HTMLEndTagCommentKind:
+      return std::make_unique<AstHtmlEndTagComment>(
+          cast<const comments::HTMLEndTagComment>(comment));
+
+    default:
+      llvm::errs() << "Unknown comment kind: " << comment->getCommentKindName() << "\n";
+      return nullptr;
+  }
+}
+
+class CommentVisitor
+    : public clang::comments::CommentVisitor<CommentVisitor, std::unique_ptr<AstDocumentation>> {
+public:
+  CommentVisitor(const clang::ASTContext& context)
+      : clang::comments::CommentVisitor<CommentVisitor, std::unique_ptr<AstDocumentation>>(),
+        m_context{context}
+  {
+  }
+
+  std::unique_ptr<AstDocumentation> visitComment(const clang::comments::Comment* comment)
+  {
+    std::unique_ptr<AstDocumentation> rv{AstDocumentation::Create(comment)};
+    for (auto child = comment->child_begin(); child != comment->child_end(); child++)
+    {
+      auto childNode = visit(*child);
+      if (childNode)
+      {
+        rv->AddChild(std::move(childNode));
+      }
+    }
+    return rv;
+  };
+
+  std::unique_ptr<AstDocumentation> visitFullComment(const clang::comments::FullComment* decl)
+  {
+    //    decl->dump(llvm::outs(), m_context);
+    std::unique_ptr<AstDocumentation> rv{AstDocumentation::Create(decl)};
+    for (auto child = decl->child_begin(); child != decl->child_end(); child++)
+    {
+      auto childNode = visit(*child);
+      if (childNode)
+      {
+        rv->AddChild(std::move(childNode));
+      }
+    }
+    return rv;
+  };
+
+  // std::unique_ptr<AstDocumentation> visitBlockCommandComment(
+  //     const clang::comments::BlockCommandComment* bc)
+  //{
+  //   auto rv{AstDocumentation::Create(bc)};
+  //   // Gather the text for the block command comment.
+  //   for (auto child = bc->child_begin(); child != bc->child_end(); child++)
+  //   {
+  //     rv->AddChild(visit(*child));
+  //   }
+  //   return rv;
+  // };
```

Comment:
Commented code?

Reply:
Removed. This was from when I did all the work in the visitor rather than in classes derived from AstDocumentation..